chromium_src/content/browser/browser_context.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "content/public/browser/browser_context.h"
+
+#include <string>
+
+#include "base/memory/ref_counted.h"
+#include "content/browser/dom_storage/dom_storage_context_wrapper.h"
+#include "content/browser/dom_storage/session_storage_namespace_impl.h"
+#include "content/browser/renderer_host/render_view_host_delegate.h"
+#include "content/public/browser/render_view_host.h"
+#include "content/public/browser/session_storage_namespace.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+
+namespace content {
+
+scoped_refptr<content::SessionStorageNamespace> CreateSessionStorageNamespace(
+    content::StoragePartition* partition,
+    const std::string& namespace_id) {
+  content::DOMStorageContextWrapper* context_wrapper =
+      static_cast<content::DOMStorageContextWrapper*>(
+          partition->GetDOMStorageContext());
+
+  return content::SessionStorageNamespaceImpl::Create(context_wrapper,
+                                                      namespace_id);
+}
+
+std::string GetSessionStorageNamespaceId(WebContents* web_contents) {
```

Comment:
this function can be moved out here if we get namespace via web_contents, navigation controller and GetSessionStorageNamespaceMap

Reply:
The issue, I think, is that the delegate definition isn't part of the public API. It's just a forward-declaration.

Reply:
in theory we can do `web_contents()->GetController().GetSessionStorageNamespaceMap()` and pick it from this by site instance ID. not sure if this is more elegant, but at least achievable via `content/public`

Reply:
Hrm. It looks like `NavigationControllerImpl::GetSessionStorageNamespace` will either get or create the default namespace. Creating the default namespace is something that can only happen with private API. Wouldn't duplicating this code mean there is a possibility that this code could be executed before the default namespace is created and be able to find the id?

Reply:
not sure, though it seems safer to keep as is, given that the file will stay with us anyways